patches/api/0468-add-BlockBreakProgressUpdateEvent.patch

```diff
@@ -0,0 +1,66 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Badbird5907 <50347938+Badbird5907@users.noreply.github.com>
+Date: Mon, 4 Mar 2024 22:18:33 -0500
+Subject: [PATCH] add BlockBreakProgressUpdateEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/block/BlockBreakProgressUpdateEvent.java b/src/main/java/io/papermc/paper/event/block/BlockBreakProgressUpdateEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..f4a020fefe4894cfe6aea5ff4e8c417181f8afd5
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/block/BlockBreakProgressUpdateEvent.java
+@@ -0,0 +1,54 @@
++package io.papermc.paper.event.block;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Entity;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.block.BlockEvent;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Range;
++
++/**
++ * Called when the progress of a block break is updated.
++ */
++public class BlockBreakProgressUpdateEvent extends BlockEvent {
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    private final float progress;
++    private final Entity entity;
++
++    @ApiStatus.Internal
++    public BlockBreakProgressUpdateEvent(@NotNull final Block block, final float progress, @NotNull final Entity entity) {
++        super(block);
++        this.progress = progress;
++        this.entity = entity;
++    }
++
++    /**
++     * The progress of the block break the progress from 0.0 - 1.0 where 0 is no damage and
++     * 1.0 is the most damaged
```

Comment:
This sentence is kinda weird (copy paste mistake?). Maybe reformat to something like
```suggestion
+     * The progress of the block break
+     * <p>
+     * The progress ranges from 0.0 - 1.0, where 0 is no damage and
+     * 1.0 is the most damaged
```

Reply:
Whoops yeah, will do